cfme/automate/import_export.py

```diff
@@ -78,38 +120,149 @@ def import_domain_from(self, branch=None, tag=None):
+                and tag == git_type_value
+            ):
                 continue
+
             # We have the domain
-            dc = self.appliance.collections.domains
-            return dc.instantiate(
-                db_id=id, name=name, description=description, git_checkout_type=git_type,
-                git_checkout_value=git_type_value)
+            return self.appliance.collections.domains.instantiate(
+                db_id=id,
+                name=name,
+                description=description,
+                git_checkout_type=git_type,
+                git_checkout_value=git_type_value,
+            )
         else:
-            raise ValueError('The domain imported was not found in the database!')
+            raise ValueError("The domain imported was not found in the database!")
+
+
+@attr.s
+class AutomateFileImport(BaseEntity):
+    file_path = attr.ib()
+
+    def import_domain_from(self, domain_from, domain_into=None, toggle_all=True):
+        """Import the domain from file using the Import/Export UI.
+
+        Args:
+            domain_from: Select domain you wish to import from
+            domain_into: Select existing domain to import into:
+            toggle_all: Select all namespaces
+
+        Returns:
+            Instance of :py:class:`cfme.automate.explorer.domain.Domain`
+        """
+        view = navigate_to(self.parent, "All")
+        view.import_file.file.fill(self.file_path)
+        view.import_file.upload.click()
+
+        file_view = self.create_view(FileImportSelectorView, wait="10s")
+        file_view.fill(
+            {"import_into": domain_into, "import_from": domain_from, "toggle_all": toggle_all}
+        )
+        file_view.commit.click()
+        return self.appliance.collections.domains.instantiate(name=domain_from)
+
+
+@attr.s
+class AutomateImportExportsCollection(BaseCollection):
+    def instantiate(
+        self,
+        import_type="git",
+        url=None,
+        username=None,
+        password=None,
+        verify_ssl=True,
+        file_path=None,
+        **kwargs
+    ):
+        if import_type == "file":
+            import_cls = AutomateFileImport
+            arg = [file_path]
+        elif import_type == "git":
+            import_cls = AutomateGitRepository
+            arg = [url, username, password, verify_ssl]
+
+        return import_cls.from_collection(self, *arg, **kwargs)
+
+    def git_repository_from_db(self, db_id):
+        """Instantiate AutomateGitRepository with db id"""
+
+        git_repositories = self.appliance.db.client["git_repositories"]
+        try:
+            url, verify_ssl = (
+                self.appliance.db.client.session.query(
+                    git_repositories.url, git_repositories.verify_ssl
+                )
+                .filter(git_repositories.id == db_id)
+                .first()
+            )
+            return self.instantiate(import_type="git", url=url, verify_ssl=verify_ssl > 0)
```

Comment:
could you explain this verify_ssl statement ?

Reply:
yup, actually my intention is just add upload file support but it was not in collection so it was difficult for me to add new class without collection means passing appliance to class :D. So I just preserved code  (current implementation) and try move in collection. All code is from old implementation except some view changes. 

```
In [21]: git_repositories = appliance.db.client["git_repositories"]      
In [22]: url, verify_ssl = (appliance.db.client.session.query(git_reposit
    ...: ories.url, git_repositories.verify_ssl).filter(git_repositories.
    ...: id == 5).first())                                               
In [23]: verify_ssl > 0                                                  
Out[23]: True
In [24]: verify_ssl                                                      
Out[24]: 1
```

Reply:
I mean you can use `bool(verify_ssl)` for instance

Reply:
@izapolsk done